packages/drivers/odsp-driver/src/createNewContainerOnExistingFile.ts

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISummaryTree } from "@fluidframework/protocol-definitions";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { UsageError } from "@fluidframework/driver-utils";
+import {
+  InstrumentedStorageTokenFetcher,
+  IOdspResolvedUrl
+} from "@fluidframework/odsp-driver-definitions";
+import { IWriteSummaryResponse } from "./contracts";
+import {
+  createCacheSnapshotKey,
+  getOrigin,
+  IExistingFileInfo,
+} from "./odspUtils";
+import { ISnapshotContents } from "./odspPublicUtils";
+import { createOdspUrl } from "./createOdspUrl";
+import { getApiRoot } from "./odspUrlHelper";
+import { EpochTracker } from "./epochTracker";
+import { OdspDriverUrlResolver } from "./odspDriverUrlResolver";
+import { convertCreateNewSummaryTreeToTreeAndBlobs, convertSummaryIntoContainerSnapshot, CreateNewContainerOnExistingFileArgs, createNewFluidContainerCore } from "./createNewUtils";
+import { ClpCompliantAppHeader } from "./contractsPublic";
+
+/**
+ * Creates a new Fluid container on an existing file.
+ */
+export async function createNewContainerOnExistingFile(
+  ...args: CreateNewContainerOnExistingFileArgs
+): Promise<IOdspResolvedUrl> {
+  const [
+    getStorageToken,
+    fileInfo,
+    logger,
+    createNewSummary,
+    epochTracker,
+    fileEntry,
+    createNewCaching,
+    forceAccessTokenViaAuthorizationHeader,
+    isClpCompliantApp
+  ] = args;
+
+  if (createNewSummary === undefined) {
+    const toThrow = new UsageError("createNewSummary must exist to create a new container");
+    logger.sendErrorEvent({ eventName: "UnsupportedUsage" }, toThrow);
+    throw toThrow;
+  }
+
+  const { id: summaryHandle } = await createNewFluidContainerOnExistingFileFromSummary(
```

Comment:
Naming seems a bit weird. Why would we want to stress "from summary", when createNewContainerOnExistingFile also creates from summary? I'd rather find a way to differentiate these two functions based on actual differences between them (and if it's hard or not important, just call the inner one "Core"